docker/vue-app/src/components/form_view/FormViewController.js

```diff
@@ -0,0 +1,390 @@
+                updateSort.push(
+                    $.ajax({
+                        type: 'POST',
+                        url: `${this.APIroot}formEditor/${item.indicatorID}/sort`,
+                        data: {
+                            sort: item.listIndex,
+                            CSRFToken: this.CSRFToken
+                        },
+                        success: () => {},
+                        error: err => console.log('ind sort post err', err)
+                    })
+                );
+            });
+            let updateParentID = [];
+            this.parentIDsToUpdate.forEach(item => {
+                updateParentID.push(
+                    $.ajax({
+                        type: 'POST',
+                        url: `${this.APIroot}formEditor/${item.indicatorID}/parentID`,
+                        data: {
+                            parentID: item.newParentID,
+                            CSRFToken: this.CSRFToken
+                        },
+                        success: () => {},
+                        error: err => console.log('ind parentID post err', err)
+                    })
+                );
+            });
+
+            const all = updateSort.concat(updateParentID);
+            Promise.all(all).then((res)=> {
+                if (res.length > 0) {
+                    this.selectNewCategory(this.formID, this.currSubformID !== null, this.selectedNodeIndicatorID);
+                }
+            });
+
+        },
+        /**
+         * adds initial sort and parentID values to app listItems object
+         * @param {Object} formNode from the Form Index listing
+         * @param {number|null} parentID parent ID of the index listing (null for form sections)
+         * @param {number} listIndex current index for that depth in the form index
+         */
+        addToListItemsObject(formNode = {}, parentID = null, listIndex = 0) {
+            const { indicatorID, sort } = formNode;
+            const item = { sort, currSortIndex: listIndex, parentID, listIndex, newParentID: '' }
+            this.listItems[indicatorID] = item;
+        },
+        /**
+         * updates the listIndex and parentID values for a specific indicator in app listItems when moved via the Form Index
+         * @param {number} indID 
+         * @param {number|null} formParIndID null for form Sections
+         * @param {number} listIndex 
+         */
+        updateListItems(indID = 0, formParIndID = 0, listIndex = 0) {
+            let item = {...this.listItems[indID]};
+            item.listIndex = listIndex;
+            item.newParentID = formParIndID;
+            this.listItems[indID] = item;
+        },
+        startDrag(evt = {}) {
+            evt.dataTransfer.dropEffect = 'move';
+            evt.dataTransfer.effectAllowed = 'move';
+            evt.dataTransfer.setData('text/plain', evt.target.id);
+        },
+        onDrop(evt = {}) {
+            evt.preventDefault();

```

Comment:
If the function gets passed a null, it will create an empty object.  So won't have a "preventDefault" method.  This will throw an error.  In this case you're likely going to not want to give it a default and instead test what is passed in to make sure it has the needed methods.

Reply:
I wrapped or chained (?.) these (depending on the situation).  I also updated evt to event to be consistent